include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_scan.h

```diff
@@ -461,51 +460,65 @@ single_pass_scan_impl(sycl::queue __queue, _InRange&& __in_rng, _OutRange&& __ou
                          [=](const sycl::nd_item<1>& item) [[intel::reqd_sub_group_size(SUBGROUP_SIZE)]]
                          {
                              auto group = item.get_group();
-                             auto local_id = item.get_local_id(0);
-                             auto stride = item.get_local_range(0);
+                             ::std::uint32_t local_id = item.get_local_id(0);
+                             constexpr ::std::uint32_t stride = wgsize;
                              auto subgroup = item.get_sub_group();
 
-                             // Obtain unique ID for this work-group that will be used in decoupled lookback
-                             TileId dynamic_tile_id(tile_id_begin);
-                             if (group.leader())
-                             {
-                                 tile_id_lacc[0] = dynamic_tile_id.fetch_inc();
-                             }
-                             sycl::group_barrier(group);
-                             std::uint32_t tile_id = tile_id_lacc[0];
+                              std::uint32_t tile_id;
+                              if constexpr (std::is_same_v<_UseDynamicTileID, ::std::true_type>)
+                              {
+                                  // Obtain unique ID for this work-group that will be used in decoupled lookback
+                                  TileId dynamic_tile_id(tile_id_begin);
+                                  if (group.leader())
+                                  {
+                                      tile_id_lacc[0] = dynamic_tile_id.fetch_inc();
+                                  }
+                                  sycl::group_barrier(group);
+                                  tile_id = tile_id_lacc[0];
+                              }
+                              else
+                              {
+                                  tile_id = group.get_group_linear_id();
+                              }
+
 
                              // Global load into local
                              auto wg_current_offset = (tile_id * elems_in_tile);
                              auto wg_next_offset = ((tile_id + 1) * elems_in_tile);
-                             size_t wg_local_memory_size = elems_in_tile;
-                             if (wg_current_offset >= n)
-                                 return;
+                             auto wg_local_memory_size = elems_in_tile;
+
                              if (wg_next_offset > n)
                                  wg_local_memory_size = n - wg_current_offset;
-
+                             _Type my_reducer{};
```

Comment:
Uses default constructor. Not an issue just now but maybe add a TODO so we know to change it.

Reply:
Good point. Added a TODO. There's a general limitation in the current version of the algorithm in that it assumes the identity is zero.